cloudformation/alarms.yaml

```diff
@@ -491,3 +491,93 @@ Resources:
       Period: 60
       Statistic: Maximum
       Threshold: 0
+  # cloudwatch synthetic support
+  MetadataCanary:
+    Type: AWS::Synthetics::Canary
+    Properties:
+      ArtifactS3Location: !Join ['', ['s3://', !Ref SyntheticsBucket, '/', !Join ['', [!Ref Stage, ig, canary]]]]
+      Code:
+        Handler: 'index.handler'
+        Script: |
+          var synthetics = require('Synthetics');
+          const log = require('SyntheticsLogger');
+
+          const apiCanaryBlueprint = async function () {
+    
+            // Handle validation for positive scenario
+            const validateSuccessful = async function(res) {
+              return new Promise((resolve, reject) => {

```

Comment:
reject is not used in the function?

Reply:
I used the API canary blueprint from cloudformation synthetics with an update to make API and Stage configurable. I'm not particularly a fan of the code in the blueprint either but it's a known quantity and there's not a ton to gain by changing a simple ping over to more stringent standards. I don't feel super strongly here so if you guys want me to clean up the code from the API canary blueprint I can.

Reply:
Yes, +1 on the API canary blueprint needs some update. 

I would suggest to clean it up here, as this will be used as our own 'blueprint' for other canary alarms to come. Cleaning it up now would set up a high standard going forward.

Reply:
pushed updates to the synthetic code